src/cli.ts

```diff
@@ -12,7 +12,7 @@ const argv = yargs.argv;
 
 (async () => {
   try {
-    const {published, reason, manifest} = await publish({args: argv._});
+    const {published, reason, manifest} = await publish({shouldCheckTag: argv.tagCheck, args: argv._});
```

Comment:
Where does ‘.tagCheck’ come from? Does that magically work when we specify ‘—no-tag-check’ without specifying a ‘yargs’ option?

Reply:
Yeah so it seems tags that begin with `--no-` are special negations in [yargs]( https://github.com/yargs/yargs/blob/f9a18bfd624a5013108084f690cd8a1de794c430/docs/tricks.md#negating-boolean-arguments)

So our `--no-tag-check` is actually setting a flag of `tag-check` to `false`. And then the kebab-case flag is automatically converted to camelCase in the options... I think?

I linked it locally to our test repo and this all seemed to work 😅 

Some tests of the main cli function would give us much more confidence.